### PR TITLE
Split interchange into exporter

### PR DESCRIPTION
**Pull request by rain-on**

### CODE REVIEW

File: slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/interchange/InterchangeV5Exporter.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  private static final int FORMAT_VERSION = 5;
+
+  private final Jdbi jdbi;
+  private final ValidatorsDao validatorsDao;
+  private final SignedBlocksDao signedBlocksDao;
+  private final SignedAttestationsDao signedAttestationsDao;
+  private final ObjectMapper mapper;
+
+  public InterchangeV5Exporter(
+      Jdbi jdbi,
```
Comment (usmansaleem) [inline slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/interchange/InterchangeV5Exporter.java:45]: nit: final modifier ??

File: slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/interchange/InterchangeV5Exporter.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+            validatorsDao
+                .findAllValidators(h)
+                .forEach(
+                    validator -> {
+                      LOG.info(
+                          "Exporting entries for validator {}",
+                          validator.getPublicKey().toHexString());
+                      try {
+                        populateValidatorRecord(h, validator, jsonGenerator);
+                      } catch (IOException e) {
```
Comment (usmansaleem) [inline slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/interchange/InterchangeV5Exporter.java:86]: nit: final ... and you can also raise `UncheckedIOException` which wraps IOException neatly.

File: slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/interchange/InterchangeV5Exporter.java [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        .findAllBlockSignedBy(h, validator.getId())
+        .forEach(
+            b -> {
+              final tech.pegasys.web3signer.slashingprotection.interchange.model.SignedBlock
+                  jsonBlock =
+                      new tech.pegasys.web3signer.slashingprotection.interchange.model.SignedBlock(
+                          b.getSlot(), b.getSigningRoot().orElse(null));
+              try {
+                mapper.writeValue(jsonGenerator, jsonBlock);
+              } catch (IOException e) {
```
Comment (usmansaleem) [inline slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/interchange/InterchangeV5Exporter.java:117]: nit: final ... any may consider `UnchecledIOException`

File: slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/interchange/InterchangeV5Exporter.java [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+        .forEach(
+            a -> {
+              final tech.pegasys.web3signer.slashingprotection.interchange.model.SignedAttestation
+                  jsonAttestation =
+                      new tech.pegasys.web3signer.slashingprotection.interchange.model
+                          .SignedAttestation(
+                          a.getSourceEpoch(), a.getTargetEpoch(), a.getSigningRoot().orElse(null));
+              try {
+                mapper.writeValue(jsonGenerator, jsonAttestation);
+              } catch (IOException e) {
```
Comment (usmansaleem) [inline slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/interchange/InterchangeV5Exporter.java:141]: nit: final

File: slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/interchange/InterchangeV5Manager.java [hunk 5]
```diff
@@ -14,136 +14,39 @@
...
-                      LOG.info(
-                          "Exporting entries for validator {}",
-                          validator.getPublicKey().toHexString());
-                      try {
-                        populateValidatorRecord(h, validator, jsonGenerator);
-                      } catch (IOException e) {
-                        throw new RuntimeException("Failed to construct a validator entry in json");
-                      }
-                    }));
+  public void importData(InputStream in) throws IOException {
```
Comment (usmansaleem) [inline slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/interchange/InterchangeV5Manager.java:43]: nit: final

**Review by usmansaleem — APPROVED** [review]
some final modifiers are missing, looks good to me otherwise.

File: slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/interchange/InterchangeV5Exporter.java (hunk 1 above)
Comment (rain-on) [reply slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/interchange/InterchangeV5Exporter.java:45]: done

File: slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/interchange/InterchangeV5Exporter.java (hunk 2 above)
Comment (rain-on) [reply slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/interchange/InterchangeV5Exporter.java:86]: done

File: slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/interchange/InterchangeV5Manager.java (hunk 5 above)
Comment (rain-on) [reply slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/interchange/InterchangeV5Manager.java:43]: done

File: slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/interchange/InterchangeV5Exporter.java (hunk 4 above)
Comment (rain-on) [reply slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/interchange/InterchangeV5Exporter.java:141]: done

**Pull request merged**